core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -147,12 +150,19 @@ abstract class AppBase(
     val eventLogInfo: Option[EventLogInfo],
     val hadoopConf: Option[Configuration]) extends Logging with CacheableProps {
 
-  var appId: String = ""
+  var appMetaData: Option[AppMetaData] = _
```

Comment:
It would be good to either have a comment of what the default value is OR assign default value instead of the placeholder.